### PR TITLE
Terminal - both Terminal and Run Command now use colors and font from prefs

### PR DESCRIPTION
**Pull request by pskowronek**
Terminal - both Terminal and Run Command now use colors and font from prefs #945 

Additionally:
- removed Shell history from prefs.
- fix for #947
- fix for #948
- fix for Main Panel look after window resized when in Terminal mode
- fixed JediTerm jar hell (multiple jars having the same classes in different versions) - one problem is still there for core and ui - but at least jars are in the same version:
```
Split package, multiple jars provide the same package:com/jediterm/terminal
Use Import/Export Package directive -split-package:=(merge-first|merge-last|error|first) to get rid of this warning
Package found in   [Jar:jediterm-core-3.17, Jar:jediterm-ui-3.17])
```

### CODE REVIEW

**Comment by pskowronek** [comment]
Btw, worth to think if colors for selected should still be there. By default jedi inverses colors, and it is a bit quicker as i see that jedi term queries for selection colors a lot, for normal they do that only once.

**Comment by pskowronek** [comment]
> However, I will check that behavior on the newest JediTerm version that's been pushed recently to muC.

The newest JediTerm still calls `Settings#getSelectionColor` way too often, I will created a bug in JediTerm repo

**Comment by pskowronek** [comment]
PR ready to review.

File: mucommander-translator/src/main/resources/dictionary_cs.properties [hunk 1]
```diff
@@ -465,8 +460,7 @@ shortcuts_dialog.quick_search.jump_to_next = Přejít na následující výslede
 shortcuts_dialog.quick_search.mark_jump_next = Označit/odznačit stávající soubor a přejít na následující výsledek hledání
 theme_editor.title = Editor témat
 theme_editor.folder_tab = Panely souborů
-theme_editor.shell_tab = Shell
-theme_editor.shell_history_tab = Historie shellu
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_cs.properties:463]: we can drop this one

File: mucommander-translator/src/main/resources/dictionary_en.properties [hunk 2]
```diff
@@ -676,8 +670,7 @@ shortcuts_dialog.quick_search.jump_to_next = Jump to next quick search result
 shortcuts_dialog.quick_search.mark_jump_next = Mark/Unmark current file and jump to next search result
 theme_editor.title = Theme editor
 theme_editor.folder_tab = Folder pane
-theme_editor.shell_tab = Shell
-theme_editor.shell_history_tab = Shell history
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_en.properties:673]: and this

File: mucommander-translator/src/main/resources/dictionary_fr.properties [hunk 3]
```diff
@@ -494,8 +489,7 @@ shortcuts_dialog.quick_search.jump_to_next = Aller au résultat suivant
 shortcuts_dialog.quick_search.mark_jump_next = Marquer/Dé-marquer le fichier courant et aller au résultat suivant
 theme_editor.title = Editeur de thèmes
 theme_editor.folder_tab = Panneau de dossier
-theme_editor.shell_tab = Shell
-theme_editor.shell_history_tab = Historique du shell
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_fr.properties:492]: and this

File: mucommander-translator/src/main/resources/dictionary_it.properties [hunk 4]
```diff
@@ -409,8 +404,7 @@ shortcuts_dialog.quick_search.jump_to_next = Salta sulla ricerca rapida seguente
 shortcuts_dialog.quick_search.mark_jump_next = Seleziona/deseleziona il file corrente e salta sulla ricerca rapida seguente
 theme_editor.title = Modifica tema
 theme_editor.folder_tab = Pannello Cartella
-theme_editor.shell_tab = Shell
-theme_editor.shell_history_tab = Shell history
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_it.properties:407]: and this

File: mucommander-translator/src/main/resources/dictionary_nl.properties [hunk 5]
```diff
@@ -463,8 +458,7 @@ shortcuts_dialog.quick_search.jump_to_next = Ga naar volgende snelzoekresultaat
 shortcuts_dialog.quick_search.mark_jump_next = Markeer/Demarkeer huidige bestand en ga naar volgend zoekresultaat
 theme_editor.title = Themabewerker
 theme_editor.folder_tab = Bestandsvenster
-theme_editor.shell_tab = Shell
-theme_editor.shell_history_tab = Shell geschiedenis
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_nl.properties:461]: and this

File: mucommander-translator/src/main/resources/dictionary_pl.properties [hunk 6]
```diff
@@ -489,8 +484,7 @@ shortcuts_dialog.quick_search.jump_to_next = Przejdź do natępnych wyników wys
 shortcuts_dialog.quick_search.mark_jump_next = Zaznacz/Odznacz aktualny plik i skocz do następnych wyników wyszukiwania
 theme_editor.title = Edytor tematu
 theme_editor.folder_tab = Panel plików
-theme_editor.shell_tab = Shell
-theme_editor.shell_history_tab = Historia shell''a
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_pl.properties:487]: and this

File: mucommander-translator/src/main/resources/dictionary_pt_BR.properties [hunk 7]
```diff
@@ -463,8 +458,7 @@ shortcuts_dialog.quick_search.jump_to_next = Pular para o próximo resultado da
 shortcuts_dialog.quick_search.mark_jump_next = Marcar/Desmarcar arquivo corrente e pular para o próximo resultado da busca
 theme_editor.title = Editor de tema
 theme_editor.folder_tab = Painel de diretório
-theme_editor.shell_tab = Shell
-theme_editor.shell_history_tab = Histórico do Shell
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_pt_BR.properties:461]: and this

File: mucommander-translator/src/main/resources/dictionary_sv.properties [hunk 8]
```diff
@@ -463,8 +458,7 @@ shortcuts_dialog.quick_search.jump_to_next = Gå till nästa snabbsöksresultat
 shortcuts_dialog.quick_search.mark_jump_next = Markera/Avmarkera aktuell fil och gå till nästa sökresultat
 theme_editor.title = Temaredigeraren
 theme_editor.folder_tab = Mappvyer
-theme_editor.shell_tab = Terminal
-theme_editor.shell_history_tab = Terminalhistorik
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_sv.properties:461]: and this

File: mucommander-translator/src/main/resources/dictionary_zh_CN.properties [hunk 9]
```diff
@@ -463,8 +458,7 @@ shortcuts_dialog.quick_search.jump_to_next = 跳至下个快速搜索的结果
 shortcuts_dialog.quick_search.mark_jump_next = 标记/取消标记当前文件并跳至下各搜索结果
 theme_editor.title = 主题编辑器
 theme_editor.folder_tab = 文件夹面板
-theme_editor.shell_tab = Shell
-theme_editor.shell_history_tab = Shell 历史记录
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_zh_CN.properties:461]: and this

File: mucommander-translator/src/main/resources/dictionary_zh_TW.properties [hunk 10]
```diff
@@ -463,8 +458,7 @@ shortcuts_dialog.quick_search.jump_to_next = 跳至下一個快速搜尋的結
 shortcuts_dialog.quick_search.mark_jump_next = 標記/取消標記目前檔案並跳至下一個搜尋結果
 theme_editor.title = 佈景主題編輯器
 theme_editor.folder_tab = Folder pane
-theme_editor.shell_tab = Shell
-theme_editor.shell_history_tab = Shell 歷史記錄
+theme_editor.terminal_tab = Terminal
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_zh_TW.properties:461]: and this

File: package/readme.txt [hunk 11]
```diff
@@ -44,6 +44,7 @@ Improvements:
 - When opening a result of a file search with text search using the internal viewer/editor, the caret is set on the first occurrence of the searched text.
 - The action categoties that are displayed in the shortcuts preferences tab are sorted by their name.
 - The row-filtering in the shortucts preferences tab now matches the specified input also against the tooltip of the row (the action's description).
+- Theme Preferences for Shell renamed to Terminal and now both Terminal and Run Command use right colors and font
```
Comment (ahadas) [inline package/readme.txt:47]: how about:
```suggestion
- Theme preferences for Shell renamed to Terminal and now both the Terminal and Run Command dialog use configured colors and font
```

**Review by ahadas — COMMENTED** [review]
looks good overall, most of the comments are about redundant entries in the dictionary files that can be dropped

File: mucommander-core/src/main/java/com/mucommander/ui/terminal/TerminalSettingsProvider.java [hunk 12]
```diff
@@ -49,6 +57,43 @@ public boolean altSendsEscape() {
...
+        return false;
+    }
+
+    @Override
+    public Font getTerminalFont() {
+        return ThemeManager.getCurrentFont(Theme.TERMINAL_FONT);
+    }
+
+    // it's never called (good, as #getTerminalFont should already return Font in the right size)
+    @Override
```
[the remark this answers is not in the document]
Comment (pskowronek) [reply mucommander-core/src/main/java/com/mucommander/ui/terminal/TerminalSettingsProvider.java:92]: for informational purposes, but I can update a comment of #getTerminalFont above to state that we don't need to override getFontSize

**Comment by pskowronek** [comment]
PR updated with PR comments

File: mucommander-core/src/main/java/com/mucommander/ui/terminal/TerminalSettingsProvider.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (ahadas) [reply mucommander-core/src/main/java/com/mucommander/ui/terminal/TerminalSettingsProvider.java]: yeah, I think that makes more sense (to update the comment of #getTerminalFont)

**Review by ahadas — APPROVED** [review]
thanks @pskowronek !

**Pull request merged**